### ISSUE TITLE
Flag to set the base config from the CLI

### ISSUE THREAD

**Issue by tjhunter** [ecmwf/WeatherGenerator#1539]
### Describe the task. It can be a feature, documentation, etc.

Currently, the `default_config.yml` is always loaded. This is an issue for example when writing integration tests. 

The goal of this task is to add the functionality to train/infer/evaluate a model from a different default config. 

Constraints:
- it must respect the current `--config` and `--options` semantics of updating the config
- only one default config can be specificied
- if no option provided, it should default to the usual `default_config.yaml`

Suggestion: have an optional flag `--base-config` that specifies the base configuration. 

### Area

- [ ] datasets, data readers, data preparation and transfer
- [ ] model
- [ ] science
- [x] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation

**Comment by kctezcan** [comment]
This is definitely useful, but does not completely address the issue of appending losses in training_config with the default user behaivour of using the default config+an overwrite config. We should decide to also use the weight of the loss as a check to see if it is active or not, whereever applicable (see also the discussion in the SW channel).

**Comment by mtar** [comment]
assign me

**Comment by clessig** [comment]
> s the issue of appending losses in training_config with the default user behaivour of using the default config+an overwrite con

I do not understand you concern: you concern is about removing a config param (e.g. a loss term) when continuing training or for fine-tuning/inference? 

Setting the weight to zero will work for `losses` but not, e.g., for `model_input` since we don't have weights there. To have a consistent way to specify this, I would suggest to have an optional config param `enabled` which defaults to True but can be added with `enabled: False` to disable sub-configs.

**Comment by grassesi** [comment]
note: This needs also be implemented in `launch_slurm.py` in the private repo.
